### PR TITLE
Remove safe mode.

### LINKED ISSUE #1433: Remove RPC safe mode
We inherited a "safe mode" from Blackcoin where RPC calls can be blocked under certain conditions. Right now, after #1432, the only time this happens is if you call `GetWarnings("rpc")` and have enabled `-testsafemode`. Without those conditions all RPC calls are allowed. Bitcoin does not have this functionality at all as it was removed in 2ae705d84178fb9faa49f92091206e92379a2c63.

### PR DESCRIPTION
**Pull request by denravonska**
This extends #1432 and closes #1433.

Bitcoin removed this in bitcoin/bitcoin@2ae705d84178fb9faa49f92091206e92379a2c63.

### CODE REVIEW

**Review by jamescowens — APPROVED** [review]
utACK

File: src/rpcserver.cpp [hunk 1]
```diff
@@ -865,12 +865,6 @@ UniValue CRPCTable::execute(const std::string& strMethod, const UniValue& params
     if (!pcmd)
         throw JSONRPCError(RPC_METHOD_NOT_FOUND, "Method not found");
 
-    // Observe safe mode
-    string strWarning = GetWarnings("rpc");
-    if (strWarning != "" && !GetBoolArg("-disablesafemode") &&
-        !pcmd->okSafeMode)
-        throw JSONRPCError(RPC_FORBIDDEN_BY_SAFE_MODE, string("Safe mode: ") + strWarning);
```
Comment (cyrossignol) [inline src/rpcserver.cpp:872]: Any reason to keep `RPC_FORBIDDEN_BY_SAFE_MODE` in rpcprotocol.h?

Comment (jamescowens) [reply src/rpcserver.cpp:872]: Looks like it is only used by

throw JSONRPCError(RPC_FORBIDDEN_BY_SAFE_MODE, string("Safe mode: ") + strWarning);

which is going away. I think it is safe to delete.

Comment (denravonska) [reply src/rpcserver.cpp:872]: Only reason is that it never occurred to me :)

**Review by cyrossignol — APPROVED** [review]

**Pull request merged**